# FAQ

####**Q: What types of data does EGA support**
	A: For sequences downloaded from NCBI, users can upload directly. If your data are downloaded elsewhere, process your data with the perl command we offer in the previous instruction before you upload them.
	If you are using a web-based EGA, the file size is limited to 30M for each uploaded file. If you are using the local VM, there is no file size limit.

####**Q: Pre-created accounts**
    A: For users' convenience, we have provided a test account stored with the data used for example tests. You can find the accounts in corresponding examples.


####**Q: Explanation of result files and file format**  
	A: For multiple genomes alignment, the result folder contains folders listed below:
	1. Genomes: Contains the information of individual files processed in this alignment, including: .fasta file after data preparation, information after repeat mask and the size analysis of the genome.
	2. Pairwise: Contains the results of pairwise alignment between reference genome and every other genomes in this alignment.
	3. jobID_mz: Combine the individual result file of pairwise alignment into .maf file by Multiz. 
	4. jobID_fasta: Convert .maf file produced in the former step into .fasta file.
	5. jobID_refined: The .fasta file in the former step may contain gaps. After refinement, the final result of alignment is stored in the refined .fasta file. 
	6. jobID_vcf: Contains a .vcf file stored with the statistical analysis of SNP and indel information in the alignment results.
	7. jobID_phylo: Contains the result files related to the phylogenetic tree, including: the final phylogenetic tree in .nwk and .phy file style, a vector graphic in .svg file style and other intermediate files produced during the construction process by RAxML. 
	
	B: For paralogous alignment, the result folder contains contents as below:
	1. Similar to the multiple genomes alignment, Genomes, Pairwise and Processing folders contain intermediate files.
	2. Results: Contains the circos graphic showing the paralogous identified by self-alignment and the final result file listing pairwise paralog requences in .fas file. Other files are intermediate files.


####**Q: hardware environment requirement of local virtual machine**  

####**Q: How to configure shared folders and mount automatically in VirtualBox **  
	A:Shared folders are used for reading and writing between VM and your laptop/desktop filesystems. To configure shared folders, follow the steps below.  

	Ensure that the VM is in a stopped state.
	Bring up the shared folders dialog by clicking the "Shared Folders" option in the VirtualBox control panel indicated by "1".

	Click the “add folder” icon indicated by "2". 

	Check the "Auto-mount" option shown by "3”.

	Choose the folder path and name indicated by “4”, then click “OK”.

	Successful configuration should lead to a dialog like this. Click “OK” to finish. 
	

